File: CPP/226.InvertBinaryTree.cpp
#include <bits/stdc++.h>
using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

void preOrderVisit(TreeNode* root) {
    if (root==nullptr) return;
    cout << root->val << " ";
    preOrderVisit(root->left);
    preOrderVisit(root->right);
}

TreeNode* invertTree(TreeNode* root) {
    if (root==nullptr) return nullptr;
    swap(root->right,root->left);
    invertTree(root->right);
    invertTree(root->left);
    return root;
}


int main() {
    TreeNode *bt1 = new TreeNode(2,new TreeNode(1),new TreeNode(3));
    //preOrderVisit(bt1);
    bt1=invertTree(bt1);
    preOrderVisit(bt1);

    cout << endl;

    TreeNode *bt2 = new TreeNode(4,new TreeNode(2,new TreeNode(1),new TreeNode(3)),new TreeNode(7,new TreeNode(6),new TreeNode(9)));
    //preOrderVisit(bt2);
    bt2=invertTree(bt2);
    preOrderVisit(bt2);
    
    return 0;
}

File: Python/226.InvertBinaryTree.py
from typing import Optional
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def printTreePre(t):
    if not t: return
    print(t.val,sep=' ')
    printTreePre(t.left)
    printTreePre(t.right)



class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root: return
        root.left,root.right=root.right,root.left
        self.invertTree(root.right)
        self.invertTree(root.left)
        return root


s=Solution()
t1=TreeNode(4,TreeNode(2,TreeNode(1),TreeNode(3)),TreeNode(7,TreeNode(6),TreeNode(9)))
printTreePre(t1)
print()
printTreePre(s.invertTree(t1))